auth/auth.go

```diff
@@ -99,6 +102,106 @@ type Token struct {
 	// Metadata  may include, but is not limited to, the body of the token
 	// response returned by the server.
 	Metadata map[string]interface{} // TODO(codyoss): maybe make a method to flatten metadata to avoid []string for url.Values
+	// TrustBoundaryData represents the trust boundary data associated with the token.
+	// It contains information about the regions or environments where the token is valid.
+	// This data is used to enforce trust boundary restrictions on requests made with the token.
+	// TrustBoundaryData is a value type for immutability after token creation.
+	TrustBoundaryData TrustBoundaryData
+}
+
+// SetAuthHeader uses the provided token to set the Authorization and trust
+// boundary headers on a request. If the token.Type is empty, the type is
+// assumed to be Bearer.
+func SetAuthHeader(token *Token, req *http.Request) {
```

Comment:
Unless this function is able to always be the **one and only** solution for setting all possible auth-related headers, it shouldn't be in this root `auth` package. If instead this is function focused on trust boundary support, it should be in a `trustboundary` subpackage.